src/components/ListItem.tsx

```diff
@@ -1,8 +1,70 @@
-import * as api from '../api';
-import './ListItem.css';
+import "./ListItem.css";
+import { updateItem, ListItem } from "../api";
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { is24HoursLater } from "../utils";
 
-type Props = Pick<api.ListItem, 'name'>;
+interface Props {
+	item: ListItem;
+	listPath: string | null;
+}
+
+export function ListItemCheckBox({ item, listPath }: Props) {
+	const [checked, setChecked] = useState(false);
+
+	const purchaseDate = item.dateLastPurchased
+		? item.dateLastPurchased.toDate()
+		: item.dateLastPurchased;
+
+	useEffect(() => {
+		if (purchaseDate) {
+			setChecked(!is24HoursLater(purchaseDate));
+		} else {
+			setChecked(false);
+		}
+	}, [item.dateLastPurchased]);
+
+	const handleCheckChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+		const isChecked = e.target.checked;
+
+		if (!isChecked && purchaseDate && !is24HoursLater(purchaseDate)) {
+			toast.error(
+				`${item.name} has already been marked as purchased in the last 24 hours.`,
+			);
+			return;
+		}
+
+		setChecked(isChecked);
```

Comment:
We probably don't want to be setting this yet, it makes error handling harder. For instance, if we ever return an error in the `if (!listPath)` condition, this is left checked despite the database not seeing an update. The user is left misinformed: the ui told them there was an error, but it also appears to have worked. It also would allow us to remove the set we make on line 48